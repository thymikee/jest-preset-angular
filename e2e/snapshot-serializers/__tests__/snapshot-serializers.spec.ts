import { TestBed } from '@angular/core/testing';

import serializer from '../../../build/serializers/ng-snapshot';
import { FooComponent } from '../foo.component';

describe('FooComponent', () => {
    test('should allow generating snapshot with removed component attributes with snapshot serializer option', () => {
        expect.addSnapshotSerializer({
            print: (val, print, indent, options, colors) =>
                serializer.print(
                    val,
                    print,
                    indent,
                    {
                        ...options,
                        omitAllCompAttrs: true,
                    },
                    colors,
                ),
            test: serializer.test,
        });
        TestBed.configureTestingModule({
            imports: [FooComponent],
        });
        const fixture = TestBed.createComponent(FooComponent);

        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
        expect(fixture.debugElement.nativeElement).toMatchSnapshot();
    });

    test('should allow generating snapshot', () => {
        expect.addSnapshotSerializer(serializer);
        TestBed.configureTestingModule({
            imports: [FooComponent],
        });
        const fixture = TestBed.createComponent(FooComponent);

        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
        expect(fixture.debugElement.nativeElement).toMatchSnapshot();
    });
});
