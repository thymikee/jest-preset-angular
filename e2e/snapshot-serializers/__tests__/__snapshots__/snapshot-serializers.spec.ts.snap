// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooComponent should allow generating snapshot 1`] = `
<foo
  condition1={[Function Boolean]}
  condition2="false"
  value1={[Function Function]}
  value2={[Function Function]}
>
  <p>
    Line 1
  </p><div>
    <div>
       val1 
    </div>
  </div>
</foo>
`;

exports[`FooComponent should allow generating snapshot 2`] = `
<div
  id="root1"
>
  <p>
    Line 1
  </p>
  <div>
    <div>
       val1 
    </div>
    
    
  </div>
</div>
`;

exports[`FooComponent should allow generating snapshot with removed component attributes with snapshot serializer option 1`] = `
<foo>
  <p>
    Line 1
  </p><div>
    <div>
       val1 
    </div>
  </div>
</foo>
`;

exports[`FooComponent should allow generating snapshot with removed component attributes with snapshot serializer option 2`] = `
<div
  id="root0"
>
  <p>
    Line 1
  </p>
  <div>
    <div>
       val1 
    </div>
    
    
  </div>
</div>
`;
